### PR TITLE
Use dynamic `year` for `Copyright` in `docusaurus.config.js`

### PR DESCRIPTION
**Pull request by jbampton**
Also update the main license to 2024

Example seen here:

https://docusaurus.io/docs/api/docusaurus-config#themeConfig

### CODE REVIEW

File: docusaurus.config.js [hunk 1]
```diff
@@ -252,7 +252,7 @@ const config = {
 		  <img src="/img/linkedin_icon.svg" alt=""/>
 		</a>
             </div>
-            <div class="footer-bottom">Copyright © 2023 The Apache
+            <div class="footer-bottom">Copyright © ${new Date().getFullYear()} The Apache
```
Comment (DaanHoogland) [inline docusaurus.config.js:255]: not sure if we should update the license this way, as above I think we need to leave the start year in place ???

Comment (jbampton) [reply docusaurus.config.js:255]: Hey @DaanHoogland do you mean from `2023 to current year` ?

Or should we use an earlier starting year from back when CloudStack started ?

Comment (DaanHoogland) [reply docusaurus.config.js:255]: no, we should use the year this website started. I am not sure if that should be 2012 (the original version) or 2023 (the docusaurus version) But the jist is that there is contents there from al those years that is copyrighted. I am not a lawyer, so I would really not know what is legal and permissable, I just know that removing an original year may enable people to "legally" copy older versions.

Comment (DaanHoogland) [reply docusaurus.config.js:255]: I think we should be safe to use `2012 - ${new Date().getFullYear()}`, though.

Comment (rohityadavcloud) [inline docusaurus.config.js:255]:
```suggestion
            <div class="footer-bottom">Copyright © 2012-${new Date().getFullYear()} The Apache
```

File: docusaurus.config.js [hunk 2]
```diff
@@ -252,7 +252,7 @@ const config = {
 		  <img src="/img/linkedin_icon.svg" alt=""/>
 		</a>
             </div>
-            <div class="footer-bottom">Copyright © 2023 The Apache
+            <div class="footer-bottom">Copyright © 2012-${new Date().getFullYear()} The Apache
```
Comment (rohityadavcloud) [inline docusaurus.config.js:255]:
```suggestion
            <div class="footer-bottom">Copyright © 2013  - ${new Date().getFullYear()} The Apache
```

File: docusaurus.config.js [hunk 3]
```diff
@@ -252,7 +252,7 @@ const config = {
 		  <img src="/img/linkedin_icon.svg" alt=""/>
 		</a>
             </div>
-            <div class="footer-bottom">Copyright © 2023 The Apache
+            <div class="footer-bottom">Copyright © 2013  - ${new Date().getFullYear()} The Apache
```
Comment (rohityadavcloud) [inline docusaurus.config.js:255]:
```suggestion
            <div class="footer-bottom">Copyright © 2012  - ${new Date().getFullYear()} The Apache
```

**Review by rohityadavcloud — APPROVED** [review]

**Pull request merged**